test/e2e/httpproxy/httpproxy_test.go

```diff
@@ -582,119 +582,100 @@ descriptors:
 				"root-ns-crd-2",
 			}
 			nonRootNamespace := "nonroot-ns-crd"
-
+	
 			BeforeEach(func() {
 				if !e2e.UsingContourConfigCRD() {
 					// Test only applies to contour config CRD.
 					Skip("")
 				}
-				for _, ns := range rootNamespaces {
-					f.CreateNamespace(ns)
-				}
-				contourConfiguration.Spec.HTTPProxy.RootNamespaces = rootNamespaces
 			})
-
+	
 			AfterEach(func() {
-				for _, ns := range rootNamespaces {
-					f.DeleteNamespace(ns, false)
-				}
+				// No need to handle namespace deletion here
 			})
-
+	
+			// Call NamespacedTest with specific namespace names
+			f.NamespacedTest(rootNamespaces[0], testRootNamespaces(rootNamespaces))
+			f.NamespacedTest(rootNamespaces[1], testRootNamespaces(rootNamespaces))
 			f.NamespacedTest(nonRootNamespace, testRootNamespaces(rootNamespaces))
```

Comment:
we only want to run this actual test once, this should rather look something like this:
```suggestion
			f.NamespacedTest(nonRootNamespace, testRootNamespaces(rootNamespaces), rootNamespaces...)
```